### PR TITLE
Amend the repository's README files

### PR DESCRIPTION
**Pull request by dpoulopoulos**
## What's changing

Simplify the README files in the repository, reducing the amount of information they contain, except for the root README file.

Retain the project description and Getting Started section in the root README file, and add links to the documentation, contributing guide, and GitHub issues.

Convert several README files into dedicated documentation pages.

Add a new CI workflow to check for broken links in Markdown and reStructuredText files.

## I already...

- [x] added some tests for any new functionality
- [x] updated the documentation

### CODE REVIEW

File: .github/workflows/notebook_integration_tests.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: SDK integration tests
+name: Tests
```
Comment (peteski22) [inline .github/workflows/notebook_integration_tests.yaml:1]:
```suggestion
name: Tests
```

Why not "Notebook integration tests"?

File: lumigator/python/mzai/backend/backend/alembic/README.md
Comment (peteski22) [inline lumigator/python/mzai/backend/backend/alembic/README.md:1]: ❗ General point on files that appear deleted now, would it have been easier to track the changes if we did `git mv` on these files before modifying them? 

e.g. the content of this file wasn't removed, it was moved/split up/etc. and now is `docs/source/operations-guide/alembic.md`

**Review by peteski22 — COMMENTED** [review]
😎 looking good @dpoulopoulos, I added two small comments.

File: lumigator/python/mzai/backend/backend/alembic/README.md
Comment (dpoulopoulos) [reply lumigator/python/mzai/backend/backend/alembic/README.md:1]: This is actually the right way to do it, you're right. I didn't think about it. I think it's kinda hard to make this change now, but I'll keep this in mind for future references. Are you okay with that?

File: .github/workflows/notebook_integration_tests.yaml (hunk 1 above)
Comment (dpoulopoulos) [reply .github/workflows/notebook_integration_tests.yaml:1]: It could be. If we have more Notebooks down the line. I'll make that change.

**Review by peteski22 — APPROVED** [review]

**Comment by dpoulopoulos** [comment]
@peteski22 yes, let's make a separate PR for this, after we merge this one.

File: README.md [hunk 2]
```diff
@@ -1,191 +1,122 @@
...
-```
-
-**Run the app locally via docker compose:**
+To uninstall Lumigator, simply run the following command:
 
 ```bash
-make local-up
-make local-logs # gets the logs from docker compose
-make local-down # shuts it down
+make stop-lumigator
```
Comment (ividal) [inline README.md:106]: I'd say "to stop Lumigator", since we're not uninstalling - or instruct how to truly purge anything Lumigator after stopping (rm the folder). WDYT?

File: README.md [hunk 3]
```diff
@@ -1,191 +1,122 @@
...
-| causal     | gpt-4o-mini, gpt-4-turbo, gpt-3.5-turbo-0125 |                 |    X    |
-| causal     | open-mistral-7b                              |                 |    X    |
+You can run and develop Lumigator locally using Docker Compose. This creates three container
+services networked together to make up all the components of the Lumigator application:
 
+- `localstack`: Local storage for datasets that mimics S3-API compatible functionality.
+- `backend`: Lumigator’s FastAPI REST API.
+- `ray`: A Ray cluster for submitting several types of jobs.
 
-### Models for Offline Evaluation
```
Comment (ividal) [inline README.md:49]: Where do users find currently supported models + tasks...? I didn't see it in a (quick) look through the other readmes + the doc - are we removing this info?

**Review by ividal — APPROVED** [review]
Thanks for the restructuring @dpoulopoulos ! A few comments to understand where we could plug in information that users might want to see at first glance vs in the docs.

File: README.md (hunk 2 above)
Comment (dpoulopoulos) [reply README.md:106]: @ividal good point. I'll rephrase it to "To stop the containers you started using Docker Compose, simply run the following command:" to make it more explicit. WDYT?

File: README.md (hunk 3 above)
Comment (dpoulopoulos) [reply README.md:49]: @ividal Yes, my plan is to move this in the docs. But first, I have to find out if this information is up to date. For example, what does it mean "Models for Offline Evaluation"? Can I pick something that is not in this list? What if I want to evaluate a model for summarization that was published yesterday? Is this an exhaustive list and if yes why?
